repos/spack_repo/builtin/packages/rivet/package.py

```diff
@@ -154,6 +158,14 @@ def configure_args(self):
             args += ["--with-hepmc3=" + self.spec["hepmc3"].prefix]
             args += ["--with-hepmc3-libpath=" + self.spec["hepmc3"].libs.directories[0]]
 
+        val = self.spec.variants["plugin-match"].value
+        if val:
+            args += [f"--with-plugin-match={' '.join(val)}"]
+
+        val = self.spec.variants["plugin-unmatch"].value
+        if val:
+            args += [f"--with-plugin-unmatch={' '.join(val)}"]
```

Comment:
Hmm, https://gitlab.spack.io/spack/spack-packages/-/jobs/17135055#L7106 indicates that this needs the version range after all.
```suggestion
        if "plugin-match" in self.spec.variants:
            val = self.spec.variants["plugin-match"].value
            if val:
                args += [f"--with-plugin-match={' '.join(val)}"]

        if "plugin-unmatch" in self.spec.variants:
            val = self.spec.variants["plugin-unmatch"].value
            if val:
                args += [f"--with-plugin-unmatch={' '.join(val)}"]
```

Reply:
The failure at https://gitlab.spack.io/spack/spack-packages/-/jobs/17135069#L1961 may be due to the default value of the new variant. It seems we are now explicitly passing `"none"` in https://gitlab.spack.io/spack/spack-packages/-/jobs/17135069#L1170.

It appears sufficient to change the `if val:` to `if val != "none":`, as applied in the suggestion above as well.

Reply:
Apparently because `val` is a tuple:
```diff
-            if val != "none":
+            if not "none" in val:
```

Reply:
Yeah, I suspected that.